__tests__/app.test.js

```diff
@@ -114,4 +115,49 @@ describe('GET /api/articles', () => {
             }
         });
     });
+});
+
+
+describe('GET /api/articles/:article_id/comments', () => {
+    test('returns array of comment objects with the given article ID and the correct properties', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then((response) => {
+            const comments = response.body.comments
+
+            expect(Array.isArray(comments)).toBe(true);
+
+            expect(comments).toBeSortedBy('created_at', {descending: true});
+
+            for(comment of comments) {
+                expect(comment).toMatchObject({
+                    comment_id: expect.any(Number),
+                    votes: expect.any(Number),
+                    created_at: expect.any(String),
+                    author: expect.any(String),
+                    body: expect.any(String),
+                    article_id: expect.any(Number),
+                });
+
+                expect(comment.article_id).toBe(1);
+            }
+        });
+    });
+    test('returns 404 error message if given correctly formatted article ID that does not exist', () => {
+        return request(app)
+        .get('/api/articles/32446254/comments')
+        .expect(404)
+        .then((response) => {
+            expect(response.body.msg).toEqual('not found');
+        });
+    });
+    test('returns 400 error message if given article ID has invalid format', () => {
+        return request(app)
+        .get('/api/articles/AHHHH/comments')
+        .expect(400)
+        .then((response) => {
+            expect(response.body.msg).toEqual('bad request');
+        });
+    });
```

Comment:
There is a very important missing test here. When given a valid article_id (2 I believe) but one that has no associated comments yet, we should expect to receive an empty array.